src/test/java/com/notably/model/block/BlockTreeTest.java

```diff
@@ -0,0 +1,104 @@
+package com.notably.model.block;
+
+import static com.notably.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
+import com.notably.model.block.exceptions.CannotModifyRootException;
+import com.notably.model.block.exceptions.DuplicateBlockException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+public class BlockTreeTest {
```

Comment:
Before your test cases, I believe you should set up a new `BlockTree` instance (basically call the `setUpTestTree` method. Similarly, after each of your test cases, I believe you should tear down the `BlockTree` instance. I believe the `@BeforeEach` and `@AfterEach` annotation should suffice.

The reason I believe you should tear down after each test case is due to some of your test cases mutating the created `BlockTree`. By setting up and tearing down the `BlockTree` after each test case, you make sure that your test cases don't affect each other's execution!